src/shared.ts

```diff
@@ -138,8 +138,22 @@ export const initStripeConnect = (
   stripePromise: Promise<StripeConnectWrapper>,
   initParams: IStripeConnectInitParams
 ): StripeConnectInstanceExtended => {
+  const clientSecretPromise = initParams.fetchClientSecret();
+  let cachedClientSecretUsed = false;
+
+  const fetchClientSecret = async () => {
```

Comment:
I also see this as lower risk, since we aren't wrapping the entire call / having that boolean with state

Reply:
If we pass in the promise as a metaOption, doesn't it still require the platform to update their SDK for it to take effect tho?

Reply:
Agree that there could be risk using the variable, e.g. if it is somehow renewed; would it make more sense for it to be a separate field in `initialize`'s param, since it's not really a meta option?

Reply:
> If we pass in the promise as a metaOption, doesn't it still require the platform to update their SDK for it to take effect tho?
Yes it does! That is correct

> Agree that there could be risk using the variable, e.g. if it is somehow renewed; would it make more sense for it to be a separate field in initialize's param, since it's not really a meta option?
`metaOptions` is meant to be a catch all for any options that aren't documented/supported for platforms. I think this performance optimization fits that category!